ouroboros-consensus/src/Ouroboros/Storage/VolatileDB/Impl.hs

```diff
@@ -276,18 +277,26 @@ reOpenDBImpl VolatileDBEnv{..} =
         st <- mkInternalStateDB _dbHasFS _dbErr _parser _maxBlocksPerFile
         return (VolatileDbOpen st, ())
 
+data GetWhat = GetBlock | GetHeader
+
 getBlockImpl :: (IOLike m, Ord blockId)
              => VolatileDBEnv m blockId
+             -> GetWhat
              -> blockId
              -> m (Maybe (SlotNo, ByteString))
-getBlockImpl env@VolatileDBEnv{..} blockId =
+getBlockImpl env@VolatileDBEnv{..} getWhat blockId =
     modifyState env $ \hasFS@HasFS{..} st@InternalState{..} ->
       case Map.lookup blockId _currentRevMap of
         Nothing -> return (st, Nothing)
         Just InternalBlockInfo {..} -> do
-          let size   = unBlockSize ibBlockSize
-              offset = AbsOffset ibSlotOffset
-          bs <- withFile hasFS ibFile ReadMode $ \hndl ->
+          bs <- withFile hasFS ibFile ReadMode $ \hndl -> do
+            let (offset, size) = case getWhat of
+                  GetBlock ->
+                    (AbsOffset $ fromIntegral ibSlotOffset, unBlockSize ibBlockSize)
+                  GetHeader ->
+                    ( AbsOffset $ fromIntegral ibSlotOffset
+                                + fromIntegral ibHeaderOffset
+                    , fromIntegral ibHeaderSize)
```

Comment:
Can you lift `AbsOffset` (and `fromIntegral`?) to a higher level (e.g., move them to the line containing `hGetExactly`)? That will hopefully allow you to put them on one line so that we can easily see the difference between the two lines.